src/sickle.ts

```diff
@@ -290,6 +280,40 @@ class Annotator {
     this.indent--;
   }
 
+  private expandSymbolsFromExportStar(exportDecl: ts.ExportDeclaration): string[] {
+    let typeChecker = this.program.getTypeChecker();
+
+    // Gather the names of local exports, to avoid reexporting any
+    // names that are already locally exported.
+    let locals =
+        typeChecker.getSymbolsInScope(this.file, ts.SymbolFlags.Export | ts.SymbolFlags.Alias);
+    let localSet: {[name: string]: boolean} = {};
+    for (let local of locals) {
+      localSet[local.name] = true;
+    }
+
+    // Expand the export list, then filter it to the symbols we want
+    // to reexport
+    let exports =
+        typeChecker.getExportsOfModule(typeChecker.getSymbolAtLocation(exportDecl.moduleSpecifier));
+    let reexports: {[name: string]: boolean} = {};
+    for (let sym of exports) {
+      let name = sym.name;
+      if (localSet.hasOwnProperty(name)) {
+        // Already exported via "export var ..." or "export {foo} from ...".
```

Comment:
'already exported' might be confused that the value is the same too, can we add the word "shadowing" to the comment.